pkg/config/validation/validation.go

```diff
@@ -3180,9 +3183,28 @@ func validateHTTPRewrite(rewrite *networking.HTTPRewrite) error {
 	if rewrite.Uri == "" && rewrite.UriRegexRewrite == nil && rewrite.Authority == "" {
 		return errors.New("rewrite must specify at least one of URI, UriRegexRewrite, or authority. Only one of URI or UriRegexRewrite may be specified")
 	}
+	if err := validateUriRegexRewrite(rewrite.UriRegexRewrite); err != nil {
+		return errors.Join(errors.New("UriRegexRewrite has errors"), err)
+	}
 	return nil
 }
 
+func validateUriRegexRewrite(regexRewrite *networking.RegexRewrite) error {
+	if regexRewrite == nil {
+		return nil
+	}
+	if regexRewrite.Match == "" || regexRewrite.Rewrite == "" {
+		return errors.New("UriRegexRewrite requires both Rewrite and Match fields to be specified")
+	}
+	var err error
+	// In the validation context we cannot directly run the Match and Rewrite strings through regexp.Compile() because
+	// they are still strings at this point. We need to verify that they would be valid RE2 regex after they are written
+	// to Envoy route, hence escape the backslashes
+	err = errors.Join(err, validateStringRegexp(strings.ReplaceAll(regexRewrite.Match, "\\", "\\\\"), "HTTPRewrite.UriRegexRewrite.Match"))
+	err = errors.Join(err, validateStringRegexp(strings.ReplaceAll(regexRewrite.Rewrite, "\\", "\\\\"), "HTTPRewrite.UriRegexRewrite.Rewrite"))
```

Comment:
I am not sure this is true. 

YAML input: 
```yaml
rewrite: "\\2"
```

In go is the string

```
rewrite := `\2`
```

Which is the yaml we want to validate against. We should pass as-is

Reply:
my bad, I see what the issue was. I was treating the Rewrite string as a regex pattern when it's actually just a string with substitution placeholders  e.g. `\2`

Reply:
ahhh... right. Thanks!!